frontend/src/reducers/chat_reducer.js

```diff
@@ -0,0 +1,17 @@
+import { GET_CHAT_TOKEN } from '../actions/chat_actions';
+import { StreamChat } from 'stream-chat'
+const chatReducer = (state = {'chat_token': '', 'chat_client': new StreamChat('kcuzfg7gnc73') }, action) => {
+    Object.freeze(state);
```

Comment:
về tìm hiểu thêm immutablejs thay vì làm như này nhé

Reply:
> về tìm hiểu thêm immutablejs thay vì làm như này nhé

https://medium.com/better-programming/3-reasons-to-avoid-using-immutablejs-with-redux-b0109d0123e8

Reply:
nhiều bài post bảo không nên dùng mà anh, hay là em làm cách khác chứ dùng thư viện này có vẻ không ổn :v

Reply:
tìm hiểu immerjs nhé